src/tests/end2end/BasicTests.cpp

```diff
@@ -32,4 +32,54 @@ TEST_P(BasicTests, BufferSetSubData) {
     EXPECT_BUFFER_U32_EQ(value, buffer, 0);
 }
 
+TEST_P(BasicTests, Buffer2Texture2Buffer) {
+
+    uint8_t data[4 * 64 * 64] = {};
+    for (unsigned int i = 0; i < 4 * 64 * 64; ++i) {
+        data[i] = i % 256;
+    }
+
+    nxt::Buffer srcBuffer = device.CreateBufferBuilder()
+        .SetAllowedUsage(nxt::BufferUsageBit::TransferDst | nxt::BufferUsageBit::TransferSrc)
+        .SetInitialUsage(nxt::BufferUsageBit::TransferDst)
+        .SetSize(4 * 64 * 64)
+        .GetResult();
+
+    nxt::Texture texture = device.CreateTextureBuilder()
+        .SetDimension(nxt::TextureDimension::e2D)
+        .SetExtent(64, 64, 1)
+        .SetFormat(nxt::TextureFormat::R8G8B8A8Unorm)
+        .SetMipLevels(1)
+        .SetAllowedUsage(nxt::TextureUsageBit::TransferDst | nxt::TextureUsageBit::TransferSrc)
+        .GetResult();
+
+    nxt::Buffer dstBuffer = device.CreateBufferBuilder()
+        .SetAllowedUsage(nxt::BufferUsageBit::TransferDst | nxt::BufferUsageBit::TransferSrc)
+        .SetInitialUsage(nxt::BufferUsageBit::TransferDst)
+        .SetSize(4 * 64 * 64)
+        .GetResult();
+
+    const uint32_t* dataView = reinterpret_cast<uint32_t*>(data);
+
+    srcBuffer.SetSubData(0, 64 * 64, dataView);
+
+    srcBuffer.FreezeUsage(nxt::BufferUsageBit::TransferSrc);
+
+    nxt::CommandBuffer commands = device.CreateCommandBufferBuilder()
+        .TransitionTextureUsage(texture, nxt::TextureUsageBit::TransferDst)
+        .CopyBufferToTexture(srcBuffer, 0, texture, 0, 0, 0, 64, 64, 1, 0)
+
+        .TransitionTextureUsage(texture, nxt::TextureUsageBit::TransferSrc)
+        .TransitionBufferUsage(dstBuffer, nxt::BufferUsageBit::TransferDst)
+        .CopyTextureToBuffer(texture, 0, 0, 0, 64, 64, 1, 0, dstBuffer, 0)
+
+        .GetResult();
+
+    queue.Submit(1, &commands);
+
+    for (unsigned int i = 0; i < 64 * 64; ++i) {
```

Comment:
Does this pass? This would create one buffer per EXPECT_BUFFER_U32_EQ xD. Can you make a helper macro that checks the content of a range of a buffer instead.